src/odemis/driver/andorcam2.py

```diff
@@ -1872,426 +1990,355 @@ def _buffer_as_array(self, cbuffer, size, metadata=None):
-        need_reinit = True
-        failures = 0
+                if ex.errno != 20024:  # DRV_NO_NEW_DATA
+                    raise  # Serious error
+            else:
+                return False  # new image!
+
+            if time.time() > tend:
+                logging.warning("Timeout after %g s", time.time() - tstart)
+                raise  # seems actually serious
+
+    def _acquire(self):
+        """
+        Acquisition thread
+        Managed via the ._genmsg Queue
+        """
         try:
-            while not self.acquire_must_stop.is_set():
-                # need to stop acquisition to update settings
-                if need_reinit or self._need_update_settings():
+            self.select()
+            has_hw_lock = False
+
+            while True: # Waiting/Acquiring loop
+                # Wait until we have a start (or terminate) message
+                self._acq_wait_start()
+
+                # Preparation & acquisition loop (until stop requested)
+                logging.debug("Preparing acquisition")
+                self._old_triggers = []  # discard all old triggers
+                synchronized = None
+                need_reconfig = True
+                failures = 0
+                while True:
+                    if self.request_hw: # TODO: check if we still need this code.
+                        need_reconfig = True # ensure we'll release the hw_lock a bit
+
+                    new_synchronized = self.data._sync_event is not None
+                    if new_synchronized != synchronized:
+                        need_reconfig = True
+                    synchronized = new_synchronized
+
+                    # Before every image, check that the camera is ready for
+                    # acquisition, with the current settings (in case they were
+                    # changed during the previous frame).
+                    if need_reconfig or self._need_update_settings():
+                        # Stop the acquisition (if already running)
+                        try:
+                            if self.GetStatus() == AndorV2DLL.DRV_ACQUIRING:
+                                self.atcore.AbortAcquisition()
+                                if has_hw_lock:
+                                    self.hw_lock.release()
+                                    has_hw_lock = False
+                                time.sleep(0.1)
+                        except AndorV2Error as ex:
+                            # it was already aborted
+                            if ex.errno != 20073: # DRV_IDLE
+                                raise
+
+                        im_res, duration, trigger_mode = self._update_settings(synchronized)
+                        if not has_hw_lock:
+                            self.hw_lock.acquire()
+                            has_hw_lock = True
+                        if not synchronized:
+                            self.atcore.StartAcquisition()
+
```

Comment:
In my opinion, such comment is not useful because it's just repeating what the code does. The code is `if trigger_mode == ...`, so it should be very clear that it checks the trigger mode.